test/services/journeyAnswers/NICsAnswersServiceImplSpec.scala

```diff
@@ -115,64 +111,35 @@ class NICsAnswersServiceImplSpec extends TableDrivenPropertyChecks with AnyWordS
     ("save a Diver related exemption", Class4ExemptionAnswers(businessId, true, Some(DiverDivingInstructor)), AnnualNonFinancials(true, Some("003")))
   )
 
-  "saveClass4BusinessData" should {
-    forAll(saveClass4DataCases) { (testDescription, answer, expectedApiData) =>
-      testDescription when {
-        "creating a new answers model when there is no existing Class 4 data" in new StubbedService {
-          override val connector: StubIFSConnector =
-            StubIFSConnector(getAnnualSummariesResult = Right(api_1803.SuccessResponseSchema(None, None, None)))
-
-          val expectedResult = buildExpectedRequestResult(expectedApiData)
-
-          service.saveClass4BusinessData(journeyCtxWithNino, answer).value.map { result =>
-            assert(result.isRight)
-            assert(connector.upsertAnnualSummariesSubmissionData === expectedResult)
-          }
-        }
-        "overriding existing details" in new StubbedService {
-          override val connector = StubIFSConnector(getAnnualSummariesResult = Right(
-            api_1803.SuccessResponseSchema(
-              None,
-              None,
-              Some(AnnualNonFinancialsType(Some(true), Some(AnnualNonFinancialsType.Class4NicsExemptionReason._006), None)))))
-
-          val expectedResult = buildExpectedRequestResult(expectedApiData)
-
-          service.saveClass4BusinessData(journeyCtxWithNino, answer).value.map { result =>
-            assert(result.isRight)
-            assert(connector.upsertAnnualSummariesSubmissionData === expectedResult)
-          }
-
-        }
-      }
-    }
-  }
-
   "saveClass4SingleBusiness" should {
     "save Class 4 journey answers when user has a single business" in new StubbedService {
-      val answers            = class4SingleBusinessAnswers
+      override val businessConnector =
+        StubIFSBusinessDetailsConnector(getBusinessesResult = api1171MultipleBusinessResponse(List(businessId)).asRight)
       override val connector = StubIFSConnector(getAnnualSummariesResult = Right(api_1803.SuccessResponseSchema(None, None, None)))
 
       val expectedApiData = AnnualNonFinancials(true, Some(TrusteeExecutorAdmin.exemptionCode))
-      StubIFSConnector(getAnnualSummariesResult = Left(SingleDownstreamError(NOT_FOUND, SingleDownstreamErrorBody.notFound)))
-      val expectedResult = buildExpectedRequestResult(expectedApiData)
+      val expectedResult  = buildExpectedRequestResult(expectedApiData)
+
+      val result = service.saveClass4SingleBusiness(journeyCtxWithNino, class4SingleBusinessAnswers).value.futureValue
 
-      service.saveClass4SingleBusiness(journeyCtxWithNino, answers).value.map { result =>
```

Comment:
`.map { result =>` passes true without testing any of the assertions in the map.
Must use `.futureValue` instead. There are many cases of this, some are fixed here, others are fixed in https://github.com/hmrc/income-tax-self-employment/pull/153

Reply:
So had to make a lot of changes to fix all the tests that would be failing but aren't actually being tested